### PR TITLE
fix: person properties ui to send only edited property

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

When updating the person properties or adding a new property we sent the current values that the UI was aware of, but if that person had been updated after loading the UI we'd send an override to a property that wasn't touched.

Ideally the API would have CRUD for person/<id>/<prop-key>, but I don't know how to do that.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

1. empty state consistent with other views
from
Screenshot 2022-10-24 at 14 26 07
to
Screenshot 2022-10-24 at 15 21 25

2. when updating or adding a property we now only send the changed/added property. 
Added a separate endpoint for this to be super explicit about the usage and so we don't accidentally start updating things we shouldn't in the future.

from
Screenshot 2022-10-25 at 14 39 41
to 
Screenshot 2022-10-25 at 14 39 29

5. Clarified what the PATCH update command does, someone during support hero looked at our api docs and was confused about what person update did.

### CODE REVIEW

File: frontend/src/scenes/persons/personsLogic.tsx [hunk 1]
```diff
@@ -197,7 +197,7 @@ export const personsLogic = kea<personsLogicType>({
                 actions.setPerson({ ...person }) // To update the UI immediately while the request is being processed
                 // :KLUDGE: Person properties are updated asynchronosly in the plugin server - the response won't reflect
                 //      the 'updated' properties yet.
-                await api.persons.update(person.id, person)
+                await api.create(`api/person/${person.id}/update_property`, { key: key, value: newValue })
```
Comment (alexkim205) [inline frontend/src/scenes/persons/personsLogic.tsx:200]: I think we moved away from building the api endpoint manually, and instead using the `ApiRequest` builder. Perhaps you can create an explicit `api.persons.updateProperty` method in `api.ts`?

File: frontend/src/lib/components/PropertiesTable.tsx [hunk 2]
```diff
@@ -313,11 +313,10 @@ export function PropertiesTable({
                     embedded={embedded}
                     dataSource={objectProperties}
                     className={className}
+                    emptyState="This person doesn't have any properties"
```
Comment (alexkim205) [inline frontend/src/lib/components/PropertiesTable.tsx:316]: nice!

**Review by alexkim205 — CHANGES_REQUESTED** [review]

File: frontend/src/scenes/persons/personsLogic.tsx (hunk 1 above)
Comment (tiina303) [reply frontend/src/scenes/persons/personsLogic.tsx:200]: the current solution was just copying `delete_property`

File: posthog/api/person.py [hunk 3]
```diff
@@ -299,6 +299,11 @@ def split(self, request: request.Request, pk=None, **kwargs) -> response.Respons
 
         return response.Response({"success": True}, status=201)
 
+    @action(methods=["POST"], detail=True)
```
Comment (macobo) [inline posthog/api/person.py:302]: Missing API docs.

Comment (tiina303) [reply posthog/api/person.py:302]: isn't that automatically generated?

Comment (macobo) [reply posthog/api/person.py:302]: For the endpoint yes, but it contains very little relevant information for users to infer how to actually use it.

See https://posthog.com/docs/api#contributing-to-api-docs for a guide on how to add more information and see other code samples in codebase on how we've documented endpoints (search for posthog.api.documentation imports)

File: frontend/src/scenes/persons/personsLogic.tsx (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/persons/personsLogic.tsx:200]: Cleared up offline: I wasn't suggesting we change the api endpoint, but rather the way we call the endpoint. `api.create` should be replaced with `api.persons.updateProperty` which uses ApiRequest builder to build up the url. There are a few examples in `api.ts::api` you can refer to but let me know if you need any help with this.

**Review by macobo — APPROVED** [review]

**Review by alexkim205 — APPROVED** [review]

**Pull request merged**